packages/terra-popup/src/_PopupContent.jsx

```diff
@@ -0,0 +1,286 @@
+  }
+
+  constructor(props) {
+    super(props);
+    this.handleClickOutside = this.handleClickOutside.bind(this);
+    this.handleKeydown = this.handleKeydown.bind(this);
+    this.handleResize = this.debounce(this.handleResize.bind(this), 100);
+  }
+
+  componentDidMount() {
+    if (this.props.closeOnEsc) {
+      document.addEventListener('keydown', this.handleKeydown);
+    }
+    if (this.props.closeOnResize) {
+      window.addEventListener('resize', this.handleResize);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.props.closeOnEsc) {
+      document.removeEventListener('keydown', this.handleKeydown);
+    }
+
+    if (this.props.closeOnResize) {
+      window.removeEventListener('resize', this.handleResize);
+    }
+  }
+
+  handleResize(event) {
+    if (this.props.closeOnResize && this.props.onRequestClose) {
+      this.props.onRequestClose(event);
+    }
+  }
+
+  handleClickOutside(event) {
+    if (this.props.closeOnOutsideClick && this.props.onRequestClose) {
+      this.props.onRequestClose(event);
+    }
+  }
+
+  handleKeydown(event) {
+    if (event.keyCode === KEYCODES.ESCAPE && this.props.onRequestClose) {
+      this.props.onRequestClose(event);
+      event.preventDefault();
+    }
+  }
+
+  debounce(fn, delay) {
+    let timer = null;
+    return (...args) => {
+      const context = this;
+      clearTimeout(timer);
+      timer = setTimeout(() => {
+        fn.apply(context, args);
+      }, delay);
+    };
+  }
+
+  render() {
+    const {
+      arrow,
+      arrowPosition,
+      children,
+      classNameInner,
+      closeOnEsc,
+      closeOnOutsideClick,
+      closeOnResize,
+      contentHeight,
+      contentHeightMax,
+      contentWidth,
+      contentWidthMax,
+      isHeaderDisabled,
+      onRequestClose,
+      refCallback,
+      ...customProps
+    } = this.props;
+
+    const showArrow = PopupContent.shouldShowArrow(arrow, arrowPosition, contentHeight, contentHeightMax, contentWidth, contentWidthMax);
+    const contentStyle = PopupContent.getContentStyle(contentHeight, contentHeightMax, contentWidth, contentWidthMax);
+    const isFullScreen = PopupContent.isFullScreen(contentHeight, contentHeightMax, contentWidth, contentWidthMax);
+
+    let content = children;
+    if (isFullScreen && !isHeaderDisabled) {
+      content = PopupContent.addPopupHeader(children, onRequestClose);
+    }
+
+    let arrowContent;
+    if (showArrow) {
+      arrowContent = arrow;
+    }
+
+    const popupContentClassNames = classNames([
+      'terra-PopupContent',
+      { 'terra-PopupContent-showArrow': showArrow },
+      { [`${CONTENT_CLASSES[arrowPosition]}`]: showArrow },
+      customProps.className,
+    ]);
+
+    const innerClassNames = classNames([
+      'terra-PopupContent-inner',
+      { 'terra-PopupContent-inner--isFullScreen': isFullScreen },
+      classNameInner,
+    ]);
+
+    // Delete the disableOnClickOutside and disableOnClickOutside prop that comes from react-onclickoutside.
```

Comment:
listed `disabledOnClickOutside and disabledOnClickOutside` instead of `disabledOnClickOutside and enabledOnClickOutside`